lib/src/parsers/oa3server-parser.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  ClassDeclaration,
+  MethodDeclaration,
+  ParameterDeclaration
+} from "ts-morph";
+import { Oa3Server, Oa3ServerVariable } from "../definitions";
+import {
+  getDecoratorConfigOrThrow,
+  getJsDoc,
+  getObjLiteralPropOrThrow,
+  getParameterPropertySignaturesOrThrow,
+  getParamWithDecorator,
+  getPropertyName,
+  getPropValueAsStringOrThrow
+} from "./parser-helpers";
+import { Oa3serverConfig } from "../syntax/oa3server";
+import { parseType } from "./type-parser";
+import { TypeTable } from "../types";
+import { LociTable } from "../locations";
+import { err, ok, Result } from "../util";
+import { OptionalNotAllowedError, ParserError } from "../errors";
+
+export function parseOa3Servers(
```

Comment:
I think we should add tests for this. There's some good examples in the `/lib/src/parsers/` directory.

Reply:
Done